### PR TITLE
refactor(lib/workers): separate get dep warning func into two diff functions

### PR DESCRIPTION
**Pull request by MaronHatoum**
## Changes
split getDepWarnings function into two different functions getDepWarningsDashboard and getDepWarningsPR

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
Looks like a bad merge commit resulted in 182 files changed:

image

**Review by rarkins — DISMISSED** [review]

File: lib/workers/repository/errors-warnings.spec.ts [hunk 1]
```diff
@@ -31,14 +31,20 @@ describe('workers/repository/errors-warnings', () => {
         "
       `);
     });
+
+    it('getWarning returns empty string', () => {
+      config.warnings = [];
+      const res = getWarnings(config);
+      expect(res).toMatchInlineSnapshot(`""`);
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.spec.ts:38]:
```suggestion
      expect(res).toBe("");
```

File: lib/workers/repository/errors-warnings.spec.ts [hunk 2]
```diff
@@ -87,6 +94,12 @@ describe('workers/repository/errors-warnings', () => {
         "
       `);
     });
+
+    it('PR warning returns empty string', () => {
+      const packageFiles: Record<string, PackageFile[]> = {};
+      const res = getDepWarningsPR(packageFiles);
+      expect(res).toMatchInlineSnapshot(`""`);
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.spec.ts:101]: same as above

File: lib/workers/repository/errors-warnings.spec.ts [hunk 3]
```diff
@@ -117,5 +130,11 @@ describe('workers/repository/errors-warnings', () => {
         "
       `);
     });
+
+    it('getError returns empty string', () => {
+      config.errors = [];
+      const res = getErrors(config);
+      expect(res).toMatchInlineSnapshot(`""`);
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.spec.ts:137]: same

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/workers/repository/errors-warnings.ts [hunk 4]
```diff
@@ -49,14 +46,30 @@ export function getDepWarnings(
...
+              ) {
+                warningFiles.push(file.packageFile);
               }
             }
           }
         }
       }
     }
+  } catch (err) {
+    // istanbul ignore next
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.ts:61]: please add a test instead

Comment (MaronHatoum) [reply lib/workers/repository/errors-warnings.ts:61]: I think we should remove the `try catch` because the catch block seems unreachable.

Comment (viceice) [reply lib/workers/repository/errors-warnings.ts]: sounds good

File: lib/workers/repository/errors-warnings.ts [hunk 5]
```diff
@@ -3,81 +3,81 @@ import type { RenovateConfig } from '../../config/types';
...
-              if (!warnings.includes(message)) {
-                warnings.push(message);
-              }
-              if (!warningFiles.includes(file.packageFile!)) {
-                warningFiles.push(file.packageFile!);
-              }
+): DepWarnings {
+  const warnings: string[] = [];
+  const warningFiles: string[] = [];
+  for (const files of Object.values(packageFiles || {})) {
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.ts:39]:
```suggestion
  for (const files of Object.values(packageFiles ?? {})) {
```

File: lib/workers/repository/errors-warnings.ts [hunk 6]
```diff
@@ -3,81 +3,81 @@ import type { RenovateConfig } from '../../config/types';
...
+    for (const file of files || []) {
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.ts:40]:
```suggestion
    for (const file of files ?? []) {
```

File: lib/workers/repository/errors-warnings.ts [hunk 7]
```diff
@@ -3,81 +3,81 @@ import type { RenovateConfig } from '../../config/types';
...
+      if (file.deps) {
+        for (const dep of file.deps || []) {
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.ts:42]:
```suggestion
        for (const dep of file.deps ?? []) {
```

File: lib/workers/repository/errors-warnings.ts [hunk 8]
```diff
@@ -3,81 +3,81 @@ import type { RenovateConfig } from '../../config/types';
...
+          if (dep.warnings?.length) {
+            const message = dep.warnings[0].message;
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.ts:44]: shouldn't we iterate over all warnings? maybe there are more than one in future

File: lib/workers/repository/errors-warnings.ts [hunk 9]
```diff
@@ -3,81 +3,86 @@ import type { RenovateConfig } from '../../config/types';
...
+              const message = w.message;
               if (!warnings.includes(message)) {
                 warnings.push(message);
               }
-              if (!warningFiles.includes(file.packageFile!)) {
-                warningFiles.push(file.packageFile!);
+              if (
+                file.packageFile &&
+                !warningFiles.includes(file.packageFile)
+              ) {
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.ts:52]:
```suggestion
              if (!warningFiles.includes(file.packageFile)) {
```

we should use a `Set<string>` instead for perf

File: lib/workers/repository/errors-warnings.ts [hunk 10]
```diff
@@ -3,81 +3,86 @@ import type { RenovateConfig } from '../../config/types';
...
-              const message = dep.warnings[0].message;
+): DepWarnings {
+  const warnings: string[] = [];
+  const warningFiles: string[] = [];
+  for (const files of Object.values(packageFiles ?? {})) {
+    for (const file of files ?? []) {
+      if (file.deps) {
+        for (const dep of file.deps ?? []) {
+          if (dep.warnings?.length) {
+            for (const w of dep.warnings) {
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.ts:44]:
```suggestion
            for (const w of dep.warnings ?? []) {
```

reduce nesting

File: lib/workers/repository/errors-warnings.ts [hunk 11]
```diff
@@ -3,81 +3,81 @@ import type { RenovateConfig } from '../../config/types';
...
-              }
-              if (!warningFiles.includes(file.packageFile!)) {
-                warningFiles.push(file.packageFile!);
-              }
+): DepWarnings {
+  const warnings: string[] = [];
+  const warningFiles: string[] = [];
+  for (const files of Object.values(packageFiles ?? {})) {
+    for (const file of files ?? []) {
+      if (file.packageFile) {
```
Comment (viceice) [inline lib/workers/repository/errors-warnings.ts:41]:
```suggestion
      // TODO: remove condition when type is fixed (#7154)
      if (file.packageFile) {
```

**Review by viceice — APPROVED** [review]

**Pull request merged**